### ISSUE TITLE
DISABLED test_profiler_with_async_rpc_udf (__main__.RpcTestWithSpawn)

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#37889]
```

May 05 15:30:34   test_profiler_with_async_rpc_udf (__main__.RpcTestWithSpawn) ... ERROR:root:Caught exception: 
May 05 15:30:34 Traceback (most recent call last):
May 05 15:30:34   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 187, in wrapper
May 05 15:30:34     fn()
May 05 15:30:34   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/dist_utils.py", line 85, in new_test_method
May 05 15:30:34     return_value = old_test_method(self, *arg, **kwargs)
May 05 15:30:34   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 888, in test_profiler_with_async_rpc_udf
May 05 15:30:34     self._profiler_test_with_rpc(RPCExecMode.ASYNC, my_sleep_func, args=(1,))
May 05 15:30:34   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 844, in _profiler_test_with_rpc
May 05 15:30:34     rpc_event = get_function_event(events, rpc_exec_mode.value)
May 05 15:30:34   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/dist_utils.py", line 231, in get_function_event
May 05 15:30:34     event = [event for event in function_events if partial_event_name in event.name][0]
May 05 15:30:34 IndexError: list index out of range
May 05 15:30:34 exiting process with exit code: 10
May 05 15:30:34 Some process exited badly, terminating rest.
May 05 15:30:34 ERROR (2.414s)
```

sample job

https://circleci.com/gh/pytorch/pytorch/5356816?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar

**Comment by rohan-varma** [comment]
Similar to https://github.com/pytorch/pytorch/issues/37557, (there is a discussion there). will disable while debugging.

**Comment by mrshenli** [comment]
fixed by #38352
